### PR TITLE
[Profiler] Remove pprof files on failure

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
Cleanup failed pprof collections.  Also atomically create/write/close/rename files so we won't have half written files there.

Also, while here, add a test that verifies that all pprof file types were created.

Depends on: #3043

### CODE REVIEW

File: module/profiler/profiler.go [hunk 1]
```diff
@@ -102,11 +102,29 @@ func (p *AutoProfiler) start() {
...
+		}
+		logger = logger.With().Str("tempFile", f.Name()).Logger()
+
+		err = p.pprof(f, prof.profileFunc)
+		if err != nil {
+			logger.Warn().Err(err).Msg("failed to generate profile")
+
+			err = os.Remove(f.Name())
+			if err != nil {
+				logger.Err(err).Msg("failed to remove profile")
```
Comment (pattyshack) [inline module/profiler/profiler.go:120]: warning is sufficient here.  "failed to remove temp profile"

File: module/profiler/profiler.go [hunk 2]
```diff
@@ -102,11 +102,29 @@ func (p *AutoProfiler) start() {
...
 		if err != nil {
-			logger.Warn().Err(err).Str("profile", prof.profileName).Msg("failed to generate profile")
+			logger.Err(err).Msg("failed to create temp profile")
+			continue
+		}
+		logger = logger.With().Str("tempFile", f.Name()).Logger()
+
+		err = p.pprof(f, prof.profileFunc)
+		if err != nil {
+			logger.Warn().Err(err).Msg("failed to generate profile")
```
Comment (pattyshack) [inline module/profiler/profiler.go:116]: This should be Err instead of Warn?

**Review by pattyshack — APPROVED** [review]

**Review by jordanschalm — APPROVED** [review]

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors cancel

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors merge

**Comment by SaveTheRbtz** [comment]
bors retry

**Pull request merged**